src/ORM/Table.php

```diff
@@ -1675,13 +1675,13 @@ protected function _transactionCommitted($atomic, $primary)
      */
     public function findOrCreate($search, callable $callback = null, $options = [])
     {
-        $options += [
+        $options = new ArrayObject($options + [
             'atomic' => true,
             'defaults' => true,
-        ];
+        ]);
 
         $entity = $this->_executeTransaction(function () use ($search, $callback, $options) {
-            return $this->_processFindOrCreate($search, $callback, $options);
+            return $this->_processFindOrCreate($search, $callback, $options->getArrayCopy());
```

Comment:
Is using `getArrayCopy()` necessary here?

Reply:
Yes, otherwise tests fail on this line:
https://github.com/cakephp/cakephp/blob/master/src/ORM/Table.php#L1892

`save()` expects an `array` for options. But depends on what is desired, not sure if eventually the `$options` argument in the `save()` function will be typehinted? If so, what will the typehint become? `ArrayObject` or `array`? And instanceof check can also be added to `save()` which then does the `getArrayCopy()`. This solution might give more flexibility, since people can then also start using `ArrayObject` in the `save()` function

Reply:
> This solution might give more flexibility, since people can then also start using ArrayObject in the save() function

Having end users need to use ArrayObject sounds like more work for them than simple arrays would be.

Reply:
`$options` for the `save()` method will remain an array, both for usability and predictability reasons.